lldb/source/Breakpoint/BreakpointIDList.cpp

```diff
@@ -20,17 +20,15 @@ using namespace lldb_private;
 
 // class BreakpointIDList
 
-BreakpointIDList::BreakpointIDList()
-    : m_invalid_id(LLDB_INVALID_BREAK_ID, LLDB_INVALID_BREAK_ID) {}
+BreakpointIDList::BreakpointIDList() : m_breakpoint_ids() {}
 
 BreakpointIDList::~BreakpointIDList() = default;
 
 size_t BreakpointIDList::GetSize() const { return m_breakpoint_ids.size(); }
 
-const BreakpointID &
-BreakpointIDList::GetBreakpointIDAtIndex(size_t index) const {
+BreakpointID BreakpointIDList::GetBreakpointIDAtIndex(size_t index) const {
   return ((index < m_breakpoint_ids.size()) ? m_breakpoint_ids[index]
-                                            : m_invalid_id);
+                                            : BreakpointID());
```

Comment:
Small question here since I'm not yet familiar with the structure of breakpoint IDs lists. Is there a scenario in which the check that `index` is within the bounds of the list fails but `BreakpointID()` is valid? Would the `BreakpointID()` always be invalid when that check fails?

Reply:
The default constructor for `BreakpointID` sets the breakpoint id and location id to invalid values.